### DISCUSSION TITLE
How do I solve "No such workflow found by id:" Exceptions?

### DISCUSSION THREAD

**Discussion by arketec** [Netflix/conductor#2834] [Q&A] [answered]
everytime I restart conductor server, I run into this:

`conductor-server_1  | 726327 [http-nio-8080-exec-7] ERROR com.netflix.conductor.core.dal.ExecutionDAOFacade [] - No such workflow found by id: b980d439-ef7b-4958-a8c5-11e357bf5a25
conductor-server_1  | 726327 [http-nio-8080-exec-7] ERROR com.netflix.conductor.rest.controllers.ApplicationExceptionMapper [] - Error ApplicationException url: '/api/tasks'
conductor-server_1  | com.netflix.conductor.core.exception.ApplicationException: No such workflow found by id: b980d439-ef7b-4958-a8c5-11e357bf5a25
conductor-server_1  | 	at com.netflix.conductor.core.dal.ExecutionDAOFacade.getWorkflowFromDatastore(ExecutionDAOFacade.java:160) ~[conductor-core-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at com.netflix.conductor.core.dal.ExecutionDAOFacade.getWorkflowModel(ExecutionDAOFacade.java:132) ~[conductor-core-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at com.netflix.conductor.core.execution.WorkflowExecutor.updateTask(WorkflowExecutor.java:1047) ~[conductor-core-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at com.netflix.conductor.service.ExecutionService.updateTask(ExecutionService.java:261) ~[conductor-core-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at com.netflix.conductor.service.TaskServiceImpl.updateTask(TaskServiceImpl.java:135) ~[conductor-core-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at com.netflix.conductor.service.TaskServiceImpl$$FastClassBySpringCGLIB$$d78c3007.invoke(<generated>) ~[conductor-core-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[conductor-es7-persistence-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:779) ~[conductor-es7-persistence-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[conductor-es7-persistence-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750) ~[conductor-es7-persistence-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at org.springframework.validation.beanvalidation.MethodValidationInterceptor.invoke(MethodValidationInterceptor.java:119) ~[conductor-es7-persistence-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[conductor-es7-persistence-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750) ~[conductor-es7-persistence-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]
conductor-server_1  | 	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692) ~[conductor-es7-persistence-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT]`

I have searched through elastic and redis to try and find the workflowId mentioned in the logs, but I cannot find it anywhere. Yet, upon each restart of conductor-server, this error appears.

Where could conductor server be getting this workflowId from?

**Comment by apanicker-nflx** [comment, marked as answer]
While traversing downt he provided stack trace ```com.netflix.conductor.service.TaskServiceImpl.updateTask(TaskServiceImpl.java:135) ~[conductor-core-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT] conductor-server_1 | at com.netflix.conductor.service.TaskServiceImpl$$FastClassBySpringCGLIB$$d78c3007.invoke(<generated>) ~[conductor-core-0.1.0-SNAPSHOT.jar!/:0.1.0-SNAPSHOT] conductor-server_1 | at```

the updateTask API is being invoked on your Conductor server with this workflow id. This API could be invoked from a worker process or another system that may be processing tasks from a previous run of your server.

**Reply by arketec** [reply]
I suspected this might be the case, but I am still learning the ins and outs of how conductor works under the hood.

thanks!
